marlowe-cli/command/Language/Marlowe/CLI/Command/Format.hs

```diff
@@ -0,0 +1,143 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+-----------------------------------------------------------------------------
+--
+-- Module      :  Language.Marlowe.CLI.Command.Format
+-- License     :  Apache 2.0
+--
+-- Stability   :  Experimental
+-- Portability :  Portable
+--
+-----------------------------------------------------------------------------
+
+-- | Format Marlowe contracts in the CLI tool.
+module Language.Marlowe.CLI.Command.Format (
+  -- * Marlowe CLI Commands
+  FormatCommand (..),
+  runFormatCommand,
+  parseFormatCommand,
+) where
+
+import Control.Monad.Except (MonadError, MonadIO (..))
+import Data.Char (toUpper)
+import Language.Marlowe.CLI.Format (
+  maybeWriteJson,
+  maybeWritePretty,
+  maybeWriteYaml,
+  readContractJson,
+  readContractPretty,
+ )
+import Language.Marlowe.CLI.Types (CliError (..))
+import Options.Applicative qualified as O
+import System.FilePath (takeExtension)
+
+-- | Marlowe CLI options for formatting contracts.
+data FormatCommand
+  = -- | Format a contract.
+    Format
+    { inputFile :: Maybe FilePath
+    -- ^ The Marlowe file containing the contract to be formatted.
+    , outputFile :: Maybe FilePath
+    -- ^ The output file for the formatted Marlowe contract.
+    , inFormat :: Maybe Format
+    -- ^ Format of the input Marlowe contract.
+    , outFormat :: Maybe Format
+    -- ^ Format of the output Marlowe contract.
+    }
+  deriving stock (Eq, Show)
+
+-- | Format a Marlowe contract.
+runFormatCommand
+  :: (MonadError CliError m)
+  => (MonadIO m)
+  => FormatCommand
+  -- ^ The command.
+  -> m ()
+  -- ^ Action for runninng the command.
+runFormatCommand Format{..} =
+  do
+    contract <- case inFormat of
+      Just Json -> readContractJson inputFile
```

Comment:
JSON is a subset of YAML, so in principle we'd just need to read YAML. This might confuse folks, however.